src/main/java/by/dragonsurvivalteam/dragonsurvival/common/handlers/HunterOmenHandler.java

```diff
@@ -0,0 +1,167 @@
+package by.dragonsurvivalteam.dragonsurvival.common.handlers;
+
+import by.dragonsurvivalteam.dragonsurvival.common.entity.creatures.DragonHunter;
+import by.dragonsurvivalteam.dragonsurvival.config.ServerConfig;
+import by.dragonsurvivalteam.dragonsurvival.registry.DSEffects;
+import by.dragonsurvivalteam.dragonsurvival.util.Functions;
+import by.dragonsurvivalteam.dragonsurvival.util.ResourceHelper;
+import com.google.common.eventbus.Subscribe;
+import it.unimi.dsi.fastutil.objects.ObjectArrayList;
+import net.minecraft.client.renderer.EffectInstance;
+import net.minecraft.core.Holder;
+import net.minecraft.core.registries.Registries;
+import net.minecraft.world.effect.MobEffectInstance;
+import net.minecraft.world.effect.MobEffects;
+import net.minecraft.world.entity.*;
+import net.minecraft.world.entity.ai.goal.target.NearestAttackableTargetGoal;
+import net.minecraft.world.entity.animal.IronGolem;
+import net.minecraft.world.entity.npc.AbstractVillager;
+import net.minecraft.world.entity.npc.Villager;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.Items;
+import net.minecraft.world.item.enchantment.Enchantment;
+import net.minecraft.world.item.enchantment.EnchantmentHelper;
+import net.minecraft.world.item.enchantment.Enchantments;
+import net.minecraft.world.level.storage.loot.parameters.LootContextParams;
+import net.neoforged.bus.api.SubscribeEvent;
+import net.neoforged.fml.common.EventBusSubscriber;
+import net.neoforged.neoforge.event.entity.EntityJoinLevelEvent;
+import net.neoforged.neoforge.event.entity.living.LivingDeathEvent;
+import net.neoforged.neoforge.event.entity.living.LivingIncomingDamageEvent;
+import net.neoforged.neoforge.event.entity.living.MobEffectEvent;
+import net.neoforged.neoforge.event.entity.player.PlayerEvent;
+
+import java.util.*;
+
+@EventBusSubscriber
+public class HunterOmenHandler {
+	// FIXME: This will fail if a player dies, then closes the game/server. The player will respawn without the effect.
+	private static final Map<UUID, MobEffectInstance> playersToReapplyHunterOmen = new HashMap<>();
```

Comment:
why not keep that in the data attachment?

Reply:
The behavior is so much simpler that I didn't feel like it really needed a data attachment. The plan was to just use the effect as a true/false sort of thing rather than progressive effects happening depending on your "threat rating". But maybe this issue with persistence after death forces me to make it a data attachment again.